### ISSUE TITLE
[Bug] XPU moe_align_block_size drops every token except index 0

### ISSUE THREAD

**Issue by TheBeaninator** [vllm-project/vllm-xpu-kernels#453]
# [Bug] XPU moe_align_block_size drops every token except index 0

## Summary
The native XPU `moe_align_block_size` op writes only the first token index into
`sorted_token_ids`; all other tokens are replaced with the padding sentinel.
Every downstream consumer of these indices (e.g. the triton wna16 fused-MoE
kernels) silently computes on 1/M of the batch, producing fluent-but-wrong
output that degrades as batch size grows.

## Repro (no model needed, ~10 lines)
```python
import torch
import vllm._custom_ops as ops

topk_ids = torch.zeros(4, 1, dtype=torch.int32, device="xpu")  # 4 tokens -> expert 0
block_size, num_experts = 16, 1
max_pad = topk_ids.numel() + num_experts * (block_size - 1)
sorted_ids = torch.empty(max_pad, dtype=torch.int32, device="xpu")
expert_ids = torch.empty(2, dtype=torch.int32, device="xpu")
num_post = torch.empty(1, dtype=torch.int32, device="xpu")

ops.moe_align_block_size(topk_ids, num_experts, block_size,
                         sorted_ids, expert_ids, num_post, None)
print(sorted_ids[:8].cpu().tolist())
```

**Expected:** `[0, 1, 2, 3, 4, 4, 4, 4]` (tokens 0–3, then padding = numel)
**Actual:** `[0, 4, 4, 4, 4, 4, 4, 4]` — tokens 1–3 are gone.

`expert_ids` and `num_tokens_post_padded` come back correct; only the token
scatter is wrong. Behavior is identical for all block sizes tested (16/32/64/128)
and any M ≥ 2.

## Impact
Any quantized-MoE path routed through the triton fused-MoE kernels on XPU is
numerically wrong (observed as garbage generation on
`Qwen3-Coder-30B-A3B-Instruct` int4). The corruption worsens with batch size
(M=1 is nearly correct because token 0 *is* written), which makes it look like
a model/quantization problem rather than an indexing bug.

## Workaround
A pure-torch replacement (stable sort by expert + exclusive-cumsum placement)
in `vllm/model_executor/layers/fused_moe/moe_align_block_size.py` restores
bit-exact behavior; with it (plus a fix for a separate `silu_and_mul` bug filed
alongside) the 30B MoE generates correctly on a Battlemage card.

## Environment
- GPU: Intel Arc Pro B70 (Battlemage G31, `Intel(R) Graphics [0xe223]`, 31 GB)
- vllm 0.24.0 (PyPI wheel), vllm-xpu-kernels 0.1.10
- torch 2.12.0+xpu, triton-xpu 3.7.1
- intel-opencl-icd 26.05.37020.3-1, libze1 1.28.2-2
- Ubuntu, kernel 7.0.0-22-generic, Python 3.12

**Comment by mayuyuace** [comment]
I tested the reproducer both on B60 and B70 with latest vllm-xpu-kernels.
I cannot reproduce your issue.
Can you try again with latest version?
https://github.com/vllm-project/vllm-xpu-kernels/releases/tag/v0.1.11

<img width="484" height="73" alt="Image" src="https://github.com/user-attachments/assets/480f2518-8119-4546-81b2-155b262e9fa4" />

**Comment by TheBeaninator** [comment]
Ahh, another fix from 0.1.11, very nice! Two more to go.
